volunteer-communication-center/functions/setup-flow.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-console, func-names */
+exports.handler = async function (context, event, callback) {
+
+  const assets = Runtime.getAssets();
+  const flowDefinition = require(assets["/studio_flow.js"].path);
+  const path = Runtime.getFunctions()['auth'].path;
+  const { getCurrentEnvironment, setEnvironmentVariable } = require(path);
+
+  const client = context.getTwilioClient();
+
+  // Deploy Twilio Studio Flow
+  function deployStudio() {
+    return configureFlowForFlex()
+      .then(() => configureFlowWithAssistant())
+      .then(() => {
+        return client.studio.flows
+        .create({
+          commitMessage: 'Code Exchange automatic deploy',
+          friendlyName: 'Volunteer Communication Center',
+          status: 'published',
+          definition: flowDefinition,
+        })
+        .then((flow) => flow)
+        .catch((err) => { throw new Error(err.details) });
+      });
+  }
+
+
+  function getWorkspace() {
+    return client.taskrouter.workspaces
+      .list({limit: 20})
+      .then(workspaces => workspaces.find(workspace => workspace["friendlyName"] == "Flex Task Assignment"))
+      .catch((err) => { throw new Error(err.details) });
+  }
+
+
+  function fetchWorkFlow(ws) {
+    return client.taskrouter.workspaces(ws)
+      .workflows
+      .list({limit: 20})
+      .then(workflows => workflows.find(workflow => workflow["friendlyName"] == "Assign to Anyone"))
+      .catch((err) => { throw new Error(err.details) });
+  }
+
+  function getChannel(ws, type) {
+    return client.taskrouter.workspaces(ws)
+      .taskChannels
+      .list({limit: 20})
+      .then(taskChannels => taskChannels.find(channel => channel["uniqueName"] == type));
+  }
+
+  async function configureFlowForFlex() {
+
+    const workspace = await getWorkspace();
+    const workflow = await fetchWorkFlow(workspace.sid);
+    const smsChannel = await getChannel(workspace.sid, "chat");
+    const voiceChannel = await getChannel(workspace.sid, "voice");
+
+    if (workspace && workflow) {
+      const states = flowDefinition["states"];
+      for (let i = 0; i < states.length; i++) {
+        if (states[i]["properties"].hasOwnProperty("workflow")) {
+          
+          states[i]["properties"]["workflow"] = workflow.sid;
+
+          if (states[i].hasOwnProperty("name") && states[i]["name"] == "send_to_flex_sms") {
+            states[i]["properties"]["channel"] = smsChannel.sid;
+          } else if (states[i].hasOwnProperty("name") && states[i]["name"] == "send_to_flex_voice") {
+            states[i]["properties"]["channel"] = voiceChannel.sid;
+          }
+        }
+      }
+      flowDefinition["states"] = states;
+
+    }
+  }
+
+
+
+  function configureFlowWithAssistant() {
+    if (process.env.AUTOPILOT_SID) {
```

Comment:
this should probably be looking for `context.AUTOPILOT_SID` before using the local process values as a fall back